main/main.ino

```diff
@@ -1613,6 +1622,65 @@ void receivingMQTT(char* topicOri, char* datacallback) {
   }
 }
 
+#ifdef MQTT_HTTPS_FW_UPDATE
+#  include <WiFiClientSecure.h>
+#  ifdef ESP32
+#    include <HTTPUpdate.h>
+#  elif ESP8266
+#    include <ESP8266httpUpdate.h>
+#  endif
+void MQTTHttpsFWUpdate(char* topicOri, JsonObject& HttpsFwUpdateData) {
+  if (strstr(topicOri, subjectFWUpdate) != NULL) {
+    if (strcmp(HttpsFwUpdateData["version"], OMG_VERSION) != 0) {
+      const char* url = HttpsFwUpdateData["url"];
+      if (!strstr((url + strlen(url - 5)), ".bin")) {
+        Log.error(F("Invalid firmware extension" CR));
+        return;
+      }
+
```

Comment:
If you are interested there is a function to stop BLE processing, stopProcessing()
so as avoid antenna usage during the OTA process

Reply:
Yep, I was thinking about this as well, adding to TODO list.